From 61585b4ad24c115033f7cfd0b5323278c2e4d799 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sun, 23 Jan 2022 22:53:35 +0100
Subject: [PATCH] Makefile.linux: Fix redundant rpath warning

By default, CHICKEN is compiled with -Wl,-rpath=/usr/lib. On Alpine,
/usr/lib is part of our default rpath and since abuild checks for
redundant rpath entries this causes a warning to be emitted.

On a side note, csc(1) (the Scheme compiler provided by CHICKEN) also
adds -Wl,-rpath=/usr/lib to the linker flags of all compiled software.
However, this only effects software compiled with CHICKEN, not CHICKEN
itself.
---
 Makefile.linux | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.linux b/Makefile.linux
index 77d59b88..f3ee1de5 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -43,8 +43,8 @@ C_COMPILER_OPTIMIZATION_OPTIONS ?= -Os -fomit-frame-pointer
 endif
 endif
 LINKER_LINK_SHARED_LIBRARY_OPTIONS = -shared
-LINKER_LINK_SHARED_DLOADABLE_OPTIONS = -L. -shared -Wl,-rpath="$(RUNTIME_LINKER_PATH)"
-LINKER_LINK_SHARED_PROGRAM_OPTIONS = -Wl,-rpath="$(RUNTIME_LINKER_PATH)"
+LINKER_LINK_SHARED_DLOADABLE_OPTIONS = -L. -shared
+LINKER_LINK_SHARED_PROGRAM_OPTIONS =
 LIBCHICKEN_SO_LINKER_OPTIONS = -Wl,-soname,lib$(PROGRAM_PREFIX)chicken$(PROGRAM_SUFFIX).so.$(BINARYVERSION)
 LIBRARIES = -lm -ldl
 NEEDS_RELINKING = yes
